core/backend_intf.ml

```diff
@@ -19,9 +19,11 @@ module type S = sig
       -> debug_print_perf_commands:bool
       -> subcommand:Subcommand.t
       -> when_to_snapshot:When_to_snapshot.t
+      -> multi_snapshot:bool
       -> trace_mode:Trace_mode.t
       -> timer_resolution:Timer_resolution.t
       -> record_dir:string
+      -> intel_pt:bool (** Select whether to use Intel PT or sampling. *)
```

Comment:
Hmm, I don't have strong opinions, but I'll lean towards no until someone asks for it. I can't really think of a case where I'd rather want higher-overhead DWARF samples over lower-overhead LBR.

Reply:
Do you have an idea about what to call this parameter? Something like trace_mode seems natural but that refers to the selection for userspace / kernel already.

Reply:
Maybe we could (in a separate, preceding commit) rename `trace_mode` to `trace_scope`? Alternatively, this could be named a `collection_mode`, because sampling isn't tracing.